backend/utils/file_storage/helpers/prompt_studio_file_helper.py

```diff
@@ -0,0 +1,96 @@
+from typing import Any, Union
+
+from file_management.exceptions import OrgIdNotValid
+from utils.file_storage.common_utils import FileStorageUtil
+from utils.file_storage.constants import FileStorageConstants, FileStorageType
+from utils.file_storage.helpers.common_file_helper import FileStorageHelper
+
+
+class PromptStudioFileHelper:
+    @staticmethod
+    def handle_sub_directory_for_prompt_studio(
+        org_id: str, user_id: str, tool_id: str, is_create: bool
+    ) -> str:
+        """Resolves a directory path meant for a user running prompt studio.
+
+        Args:
+            org_id (str): Organization ID
+            user_id (str): User ID
+            tool_id (str): ID of the prompt studio tool
+            is_create (bool): Flag to create the directory
+
+        Returns:
+            str: The absolute path to the directory meant for prompt studio
+        """
+        if not org_id:
+            raise OrgIdNotValid()
```

Comment:
@harini-venkataraman this would be raised only when `org_id` is an empty string. From the type hint, it cannot even be `None` so having a separate `OrgIdNotValid` exception doesn't seem useful. Ideally all `org_id` related validation should have happened earlier in the program flow but having it here causes no harm. However let's raise an explicit `APIError` indicating that `org_id` is empty and remove any extra exceptions defined.

Reply:
I have removed the entire check from the method used to get or create sub directory. This check is to be done at the point of invocation.